packages/server-wallet/src/message-service/test-message-service.ts

```diff
@@ -85,7 +148,9 @@ export class TestMessageService implements MessageServiceInterface {
 
           await delay(delayAmount, {signal: this._abortController.signal});
         }
-        await this._handleMessages(messages);
+        if (!this._destroyed) {
+          this.emit('deliveryRequested', messages);
```

Comment:
Curious, what is the benefit of using events here? Can we just call some function?

Reply:
Events are used here to communicate between the two different message services.

I think we could use callbacks but I think it's also kind of clunky:
```
const messageService1 = new MessageService();
const messageService2 = new MessageService();
// We need to register callbacks with eachother to trigger message delivery
messageService1.deliverMessage = m=>messageService2.handleMessage(m);
messageService2.deliverMessage = m=>messageService1.handleMessage(m);
```